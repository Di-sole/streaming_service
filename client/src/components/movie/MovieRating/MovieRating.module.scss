.container {
    width: 100%;
    margin-top: 24px;
    display: flex;
    align-items: center;
    padding: 10px;
    border-radius: 8px;
    background-color: rgba(255, 255, 255, 0.08);
    cursor: pointer;

    &__value {
        display: flex;
        width: 64px;
        min-height: 64px;
        height: 28px;
    }

    &__textContainer {
        margin-left: 16px;
        font-size: 15px;
        line-height: 20px;
        font-weight: 400;
        color: #ffffffb8;
        text-align: start;
    }

    &__text {
        &_rating {
            font-weight: 600;
            color: #fff;
        }

        &_marks {
            font-size: 13px;
            margin-top: 4px;
            line-height: 16px;
        }
    }

    &__estimate {
        margin-left: auto;
        border: 1px solid #ffffff52;
        width: 72px;
        height: 24px;
        border-radius: 6px;
        font-size: 13px;
        line-height: 16px;
        font-weight: 500;
        color: #fff;
        padding: 3px 7px;
    }
}

@media (max-width: 1159px) {
    .container {
        margin-top: 36px;
        margin-bottom: 32px;

        &__value {
            max-width: 48px;
            min-height: 48px;
        }

        &__text {
            &_marks {
                display: none;
            }
        }
    }
}

@media (max-width: 879px) {
    .container {
        margin-top: 24px;
    }
}

@media (max-width: 599px) {
    .container {
        margin-top: 16px;
        &__value {
            min-width: 48px;
        }
    }
}

@media (max-width: 300px) {
    .container {
        &__textContainer {
            margin-left: 8px;
        }
    }
}
