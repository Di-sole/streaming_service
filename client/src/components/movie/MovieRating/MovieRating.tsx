import styles from './MovieRating.module.scss';
import TextSquareUI from '../../UI/squares/TextSquareUI/TextSquareUI';
import { useTranslation } from 'next-i18next';
import IMovie from '@/models/IMovie';
import { declineWord } from '../../../utils/functions';
import BorderedButton from '@/components/UI/buttons/BorderedButton/BorderedButton';

interface MovieRatingProps {
    movie: IMovie;
}

const MovieRating = ({ movie }: MovieRatingProps) => {
    const { t } = useTranslation('movie');
    return (
        <div className={styles.container}>
            <div className={styles.container__value}>
                <TextSquareUI value={Number(movie.ratingKp.toFixed(1))} textSize="medium" />
            </div>

            <div className={styles.container__textContainer}>
                <h2 className={[styles.container__text, styles.container__text_rating].join(' ')}>
                    {t('movieRating')}
                </h2>
                <p className={[styles.container__text, styles.container__text_plot].join(' ')}>{t('hightlights')}</p>
                <p className={[styles.container__text, styles.container__text_marks].join(' ')}>
                    {`${new Intl.NumberFormat('ru-RU').format(movie.votesKp)} ${declineWord(movie.votesKp, [
                        t('userRate.0'),
                        t('userRate.1'),
                        t('userRate.2'),
                    ])}`}
                </p>
            </div>

            <BorderedButton className={styles.container__estimate} size='small'>{t('rate')}</BorderedButton>
        </div>
    );
};

export default MovieRating;
