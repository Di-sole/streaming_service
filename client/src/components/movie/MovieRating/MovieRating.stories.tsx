import { Meta, StoryObj } from '@storybook/react';
import MovieRating from './MovieRating';
import './MovieRating.module.scss';

import { INITIAL_VIEWPORTS } from '@storybook/addon-viewport';
import IMovie from '@/models/IMovie';

const meta: Meta<typeof MovieRating> = {
    title: 'Information/Rating',
    component: MovieRating,
    parameters: {
        viewport: {
            viewports: INITIAL_VIEWPORTS,
            default: 'desktop',
        },
    },
};

export default meta;

type Story = StoryObj<typeof MovieRating>;

const data: IMovie = {
    id: 1,
    type: 'movie',
    name: 'Название',
    enName: 'Название',
    posterUrl: '/img/posterTest.jpg',
    posterPreviewURL: '/img/posterTest.jpg',
    year: 1900,
    description: '',
    shortDescription: '',
    ageRating: 0,
    ratingKp: 9.5,
    votesKp: 11111,
    movieLength: 126,
    genres: [],
    countries: [],
    persons: [],
    trailer: '',
    watchingWithMovie: [],
    comments: [],
};

export const Rating: Story = {
    decorators: [
        (Story) => (
            <div style={{ background: '#100e19', width: '400px' }}>
                <div style={{ margin: '10px' }}>
                    <Story />
                </div>
            </div>
        ),
    ],
    args: {
        movie: data,
    },
};
