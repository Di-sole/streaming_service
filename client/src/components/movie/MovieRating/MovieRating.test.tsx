import { render, screen } from '@testing-library/react';
import MovieRating from './MovieRating';
import { movieTest } from '../../../testAsserts/testItems';

jest.mock('next-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

describe('MOVIERATING SNAPSHOTS TESTS', () => {
    test('MovieRating should not be changed', () => {
        render(<MovieRating movie={movieTest} />);

        const div = screen.getByTestId('movieRating');
        expect(div).toMatchSnapshot();
    });
});
