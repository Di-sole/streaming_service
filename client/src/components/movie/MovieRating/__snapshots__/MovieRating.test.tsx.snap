// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MOVIERATING SNAPSHOTS TESTS MovieRating should not be changed 1`] = `
<div
  class="container"
  data-testid="movieRating"
>
  <div
    class="container__value"
  >
    <div
      class="value value_medium value_high"
      data-testid="textSquare"
    >
      8.8
    </div>
  </div>
  <div
    class="container__textContainer"
  >
    <h2
      class="container__text container__text_rating"
    >
      movieRating
    </h2>
    <p
      class="container__text container__text_plot"
    >
      hightlights
    </p>
    <p
      class="container__text container__text_marks"
    >
      1 580 297 userRate.2
    </p>
  </div>
  <button
    class="button button_small container__estimate"
    type="button"
  >
    rate
  </button>
</div>
`;
